watershed_workflow/hilev.py

```diff
@@ -977,14 +980,14 @@ def my_refine_func(*args):
         return any(rf(*args) for rf in refine_funcs)        
 
     if mesh_rivers:
-        rivers_tri = rivers
-    else:
-        rivers_tri = None
-    vertices, triangles = watershed_workflow.triangulation.triangulate(hucs, rivers_tri,
-                                                             tol=tol, verbose=verbose,
-                                                             refinement_func=my_refine_func,
-                                                             min_angle=refine_min_angle,
-                                                             enforce_delaunay=enforce_delaunay)
+        allow_boundary_steiner=False
```

Comment:
Can we change this logic?  Just because one wants to mesh the river does not necessarily mean they don't want Steiner points on the river.  Maybe set this if river_corrs is not None?

Reply:
I am confused. May be there is different understanding of what "mesh_river" option entail. Currently I have based logic off the understanding that  "mesh_river" == mixed-element mesh. 
Are we considering the scenario where even in the triangulation, we would want to honor NHDFlowlines by placing nodes on them? If yes, I would recommend different flag for that.

Reply:
So are we set on the following logic?

- if river_corr exists, we do the mixed-element mesh
- if mesh_river is true, do a TIN with the river corridor
- otherwise do a standard TIN where the river corridor is just used for refinement

If that gets too complicated, we could turn "mesh_river" into a string instead of a bool, and have 3 options (e.g. mesh_river=False, mesh_river="TIN", mesh_river="mixed element" or something like that.

Reply:
I went with the first option.